src/bindings/wasm/tvgWasmLottieAnimation.cpp

```diff
@@ -85,8 +133,20 @@ class __attribute__((visibility("default"))) TvgLottieAnimation
             return false;
         }
 
-        //back up for saving
-        this->data = data;
+        this->data = data; //back up for saving
+
+        if (engine == tvg::CanvasEngine::Wg) {
```

Comment:
I think this WG initialization should be at creation time. Please double-check this.
```
explicit TvgLottieAnimation(string engine = "sw", string selector = "")
```